src/sentry/incidents/tasks.py

```diff
@@ -181,3 +183,34 @@ def auto_resolve_snapshot_incidents(alert_rule_id, **kwargs):
         auto_resolve_snapshot_incidents.apply_async(
             kwargs={"alert_rule_id": alert_rule_id}, countdown=1
         )
+
+
+@instrumented_task(
+    name="sentry.incidents.tasks.process_pending_incident_snapshots", queue="incident_snapshots"
+)
+def process_pending_incident_snapshots():
+    """
+    Processes PendingIncidentSnapshots and creates a snapshot for any snapshot that
+    has passed it's target_run_date.
+    """
+    from sentry.incidents.logic import create_incident_snapshot
+
+    batch_size = 50
+
+    now = timezone.now()
+    pending_snapshots = PendingIncidentSnapshot.objects.filter(
+        target_run_date__lte=now
+    ).select_related("incident")
+
+    if not pending_snapshots:
+        return
+
+    processed = 0
+    for pending_snapshot in pending_snapshots:
+        incident = pending_snapshot.incident
+        if processed > batch_size:
+            process_pending_incident_snapshots.apply_async(countdown=1)
+            break
+        else:
+            create_incident_snapshot(incident, windowed_stats=True)
```

Comment:
We should probably skip creating the snapshot if the incident isn't still closed. Either here, or inside `create_incident_snapshot`